src/change_processor.rs

```diff
@@ -124,46 +125,20 @@ impl JobThreadContext {
 
         // For a given VFS event, we might have many changes to different parts
         // of the tree. Calculate and apply all of these changes.
-        let applied_patches = {
-            let mut tree = self.tree.lock().unwrap();
-            let mut applied_patches = Vec::new();
-
-            match event {
-                VfsEvent::Create(path) | VfsEvent::Write(path) | VfsEvent::Remove(path) => {
-                    // Find the nearest ancestor to this path that has
-                    // associated instances in the tree. This helps make sure
-                    // that we handle additions correctly, especially if we
-                    // receive events for descendants of a large tree being
-                    // created all at once.
-                    let mut current_path = path.as_path();
-                    let affected_ids = loop {
-                        let ids = tree.get_ids_at_path(&current_path);
-
-                        log::trace!("Path {} affects IDs {:?}", current_path.display(), ids);
-
-                        if !ids.is_empty() {
-                            break ids.to_vec();
-                        }
-
-                        log::trace!("Trying parent path...");
-                        match current_path.parent() {
-                            Some(parent) => current_path = parent,
-                            None => break Vec::new(),
-                        }
-                    };
-
-                    for id in affected_ids {
-                        if let Some(patch) = compute_and_apply_changes(&mut tree, &self.vfs, id) {
-                            if !patch.is_empty() {
-                                applied_patches.push(patch);
-                            }
-                        }
-                    }
+        let applied_patches = match event {
+            VfsEvent::Write(path) => {
+                if path.is_dir() {
```

Comment:
Are there any docs that talk about what circumstances will cause `Write` to be emitted for a directory?

Reply:
I see no docs about but out of my testing it seems to happen every time a file or folder gets added or removed from the specific folder the parent folders are not effected.